src/lib/storage/frame_of_reference_segment.cpp

```diff
@@ -88,6 +88,7 @@ std::optional<CompressedVectorType> FrameOfReferenceSegment<T, U>::compressed_ve
 }
 
 template class FrameOfReferenceSegment<int32_t>;
-template class FrameOfReferenceSegment<int64_t>;
+// int64_t disabled for now, as vector compression cannot handle 64 bit values
```

Comment:
Vector compression in itself can handle 64-bit, but FoR only makes sense with bit packing, correct? That should be clarified here.

Reply:
No, vector compression does not support larger integers than 4 byte integers. Since we compress dictionary offsets in dictionary-encoded segments, that was never a problem.
But FoR (more or less) compresses the actual values.

And yes, FoR does not have an advantage when the deltas cannot be compressed via null surpression (~bit packing).

Reply:
Would change to:
```
// Support for int64_t is disabled for now, because the current vector compression used by
// the FrameOfReverence encoding can only handle vectors of uint32_t. As the delta values for
// long segments can be larger than uint32_t, larger than 4-byte-integers are not supported.
// template class FrameOfReferenceSegment<int64_t>;
```

Would that address all your issues?

Reply:
Whoopsie ... already merged. :)

There will be another PR today (table generator does not accept SegmentEncodingSpecs). I would add this comment there, if you find it helpful.

Reply:
Sounds good